File: src/main/java/io/stouder/adonis/service/impl/AdonisRouteServiceImpl.java
package io.stouder.adonis.service.impl;

import com.intellij.openapi.project.Project;
import io.stouder.adonis.cli.json.routes.RouteDomain;
import io.stouder.adonis.service.AdonisAceService;
import io.stouder.adonis.service.AdonisRouteService;

import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.function.Consumer;

public class AdonisRouteServiceImpl implements AdonisRouteService {
    private final Project project;

    public AdonisRouteServiceImpl(Project project) {
        this.project = project;
    }

    @Override
    public void fetchRoutes(Consumer<Map<String, Optional<RouteDomain[]>>> callback) {
        AdonisAceService adonisAceService = AdonisAceService.getInstance(this.project);
        adonisAceService.runAceGetCommandAsyncOnEveryRoots(callback, List.of("list:routes", "--json"), RouteDomain[].class);
    }

}

File: src/main/java/io/stouder/adonis/service/impl/AdonisRouteServiceImpl.kt
package io.stouder.adonis.service.impl

import com.intellij.openapi.project.Project
import io.stouder.adonis.cli.json.routes.RouteDomain
import io.stouder.adonis.service.AdonisAceService
import io.stouder.adonis.service.AdonisRouteService
import java.util.*

class AdonisRouteServiceImpl(private val project: Project) : AdonisRouteService {

    override fun fetchRoutes(callback: (Map<String, Optional<Array<RouteDomain>>>) -> Unit) {
        val adonisAceService = AdonisAceService.getInstance(project)
        adonisAceService.runAceGetCommandAsyncOnEveryRoots(callback, listOf("list:routes", "--json"), Array<RouteDomain>::class.java)
    }
}